template.tpl

```diff
@@ -453,173 +453,145 @@ const url = 'https://content.zeotap.com/sdk/zeotap.min.js';
-      var userProperties = getUserpropertiesFromData(eventData,propertiesList);
-      callInWindow('zeotap.callMethod', 'setEventProperties',data.user_logout,{});
+    if (data.loginid_exists) {
+      identities.loginid = data.loginid;
     }
-    
-    else if( matchStringWithRegex(eventData[eventNameKey],regex) || isNamePresentIn(eventList,eventData[eventNameKey])){
-      log('regex matched with event name');
-      log('zeotap.callMethod', 'setEventProperties',eventData[eventNameKey],data_wo_pii,extraProperties);
-      //assuming setPageProperties( ...args) 
-      callInWindow('zeotap.callMethod', 'setEventProperties',eventData[eventNameKey],data_wo_pii,extraProperties);
-
+    if (data.cellno_exists) {
+      identities.cellno_cc = data.cellno;
     }
-    
+    callInWindow('zeotap.callMethod', 'setUserIdentities', identities, data.areIdentitiesHashed);
+
+  } else if (eventData[eventNameKey] == data.user_logout) {
+    callInWindow('zeotap.callMethod', 'unsetUserIdentities');
+    const propertiesList = data.user_attributes;
+    const userProperties = getUserpropertiesFromData(eventData, propertiesList);
+    callInWindow('zeotap.callMethod', 'setEventProperties', data.user_logout, {});
+  } else if (matchStringWithRegex(eventData[eventNameKey], regex) || isNamePresentIn(eventList, eventData[eventNameKey])) {
+    log('regex matched with event name');
+    log('zeotap.callMethod', 'setEventProperties', eventData[eventNameKey], data_wo_pii, extraProperties);
+    //assuming setPageProperties( ...args) 
+    callInWindow('zeotap.callMethod', 'setEventProperties', eventData[eventNameKey], data_wo_pii, extraProperties);
   }
 }
-  
 
+// main execution
+if (zeotapCallMethod == undefined) {
+  const consentOptions = {
+    'default': [{ key: 'useConsent', value: false }],
+    'tcf': [{ key: 'useConsent', value: true }, { key: 'checkForCMP', value: true }],
+    'custom': [{ key: 'useConsent', value: true }, { key: 'checkForCMP', value: false }]
+  };
+  const consentMethod = data.consent_method;
+  const options = {
+    user_country: data.user_country,
+    allowIDP: data.allowIDP,
+    partnerId: data.partnerId,
+    useConsent: data.consent_method === 'default' ? false : true,
+    checkForCMP: data.consent_method === 'tcf' ? true : false,
+    allowGAClientId: data.allowGAClientId,
+    gaClientIdCookiePrefix: data.gaClientIdCookiePrefix,
+    gaUserIdCookieName: data.gaUserIdCookieName,
+    gaUserIdOnlyLoginEvent: data.gaUserIdOnlyLoginEvent
+  };
+
+  setInWindow("zeotap", { _q: [], _qcmp: [] });
+  setInWindow('zeotap.callMethod', function () {
+    callInWindow('zeotap._q.push', arguments);
+  });
+  log('zeotap.callMethod', 'init', data.writeKey, options);
+  callInWindow('zeotap.callMethod', 'init', data.writeKey, options);
+  injectScript(url, data.gtmOnSuccess, data.gtmOnFailure, 'zeoCollect');
+}
 
-injectScript(url, data.gtmOnSuccess, data.gtmOnFailure, 'zeoCollect');
+if (!!dataLayer) {
```

Comment:
Do we need to check for dataLayer array length as well?

Reply:
noted